var/spack/repos/builtin/packages/regcm/package.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Regcm(AutotoolsPackage):
+    """RegCM ICTP Regional Climate Model."""
+
+    homepage = 'https://gforge.ictp.it/gf/project/regcm/'
+
+    version('4.7.0', sha256='456631c10dcb83d70e51c3babda2f7a1aa41ed9e60cb4209deb3764655267519',
+            url='https://gforge.ictp.it/gf/download/frsrelease/259/1845/RegCM-4.7.0.tar.gz')
+
+    variant('debug', default=False,
+            description='Build RegCM using debug options.')
+    variant('profile', default=False,
+            description='Build RegCM using profiling options.')
+    variant('singleprecision', default=False,
+            description='Build RegCM using single precision float type.')
+
+    variant('knl', default=False,
```

Comment:
(FYI: no change required, but you may be interested) I assume that only one of these is set for a given architecture? These could be bundled into a single variant which enforces that only one can be chosen. Also: ideally, these could be determined automatically by examining `self.spec.architecture.target`, vs. requiring the user to specify the appropriate variant.

Reply:
Actually, if the compiler used is intel/pgi/others, the flags can be added together to create a fat binary for multiple archs, while on gcc it's not allowed, so the last set would be the only arch produced. I'm going to use a single variant to enforce a single architecture is built, if you agree.

Reply:
> these could be determined automatically by examining self.spec.architecture.target, vs. requiring the user to specify the appropriate variant.

When printing `self.spec.architecture` I get `linux-centos7-x86_64`, while `self.spec.architecture.target` returns `x86_64`. I tried digging in `self.spec.architecture`, but I found only generic architecture details.
These flags are meant to add instructions for specific Intel families, to take advantage of available extensions to the x86_64 architecture.
Moreover, the user may want to build from a different machine from the final one.
In any case, if not provided, a "classic" x86_64 executable is built, without any extension support.

So I'm pushing a new commit with a single multi-value variant for these extension flags.